lib/sync.js

```diff
@@ -105,7 +105,7 @@ module.exports = function resolveSync(x, options) {
         if (n) return maybeRealpathSync(realpathSync, n, opts);
     }
 
-    var err = new Error("Cannot find module '" + x + "' from '" + parent + "'");
+    var err = new Error("Cannot find module '" + x + "' from '" + parent + "'. Please verify that module installed, the package.json has 'main' field or module has index.js file");
```

Comment:
I'm a bit unclear what this is saying. What module should the user verify?

Reply:
The one is not found

Reply:
ok - why is "installed" here? obviously you can't resolve anything that's not installed.

Reply:
I can remove this obvious part: `Please verify that the module's package.json has 'main' field or it has index.js file`

Reply:
I guess I’m wondering why all of this isn’t obvious. Modules must be installed to be resolved, and if it doesn’t have a “main” (explicit or implicit) then there’s nothing to resolve in the first place.

Reply:
Will it be enough?

Reply:
As you can see on my example, it isn't obvious)
The case is: I have an installed module [stretch-layout](https://www.npmjs.com/package/stretch-layout). It has neither index.js, nor `main` field in package.json. But it has `module` field in package.json (which is not even in spec yet) and it seems enough for bundler to resolve and build this module. And my point is that error "Cannot find module" is not precise enough, because there is module. But it's not prepared accurately. I think it's a bad pattern for that module, but it would be great if libraries like `resolve`, which are used by many other libraries, could detect error more precisely or even add support for `module` field, but it seems like more discussional change.

Reply:
The `module` field is both not standard, and also never ever will be - it's something a couple bundlers use that they never should have, and that no tooling should support by default.

Indeed, that library is broken, and given that it doesn't even have a repo link, a "main", or "exports", i'd suggest avoiding it entirely.

While I agree a better error message here might be helpful, it'd be being shown to the consumer, who has no ability to fix the situation - only to perhaps file an issue. That's not going to help you in this case, because this package has no way to do so.